object/oidc_discovery.go

```diff
@@ -160,3 +172,43 @@ func GetJsonWebKeySet() (jose.JSONWebKeySet, error) {
 
 	return jwks, nil
 }
+
+func GetWebFinger(resource string, rels []string, host string) (WebFinger, error) {
+	wf := WebFinger{}
+
+	resourceSplit := strings.Split(resource, ":")
+
+	if len(resourceSplit) != 2 {
+		return wf, fmt.Errorf("invalid resource")
+	}
+
+	resourceType := resourceSplit[0]
+	resourceValue := resourceSplit[1]
+
+	oidcDiscovery := GetOidcDiscovery(host)
+
+	switch resourceType {
+	case "acct":
+		user, err := GetUserByEmailOnly(resourceValue)
+		if err != nil {
+			return wf, err
+		}
+
+		if user == nil {
+			return wf, fmt.Errorf("user not found")
+		}
+
+		wf.Subject = resource
+
+		for _, rel := range rels {
+			if rel == "http://openid.net/specs/connect/1.0/issuer" {
```

Comment:
why rel must be `"http://openid.net/specs/connect/1.0/issuer"` ?

Reply:
is there any reference?

Reply:
![image](https://github.com/user-attachments/assets/afe91fc0-20d5-4d6e-81e7-f378c12d4562)
![image](https://github.com/user-attachments/assets/7b77a6bd-ddbe-4c6a-bc04-d524572a6619)

Section 3.1 in the RFC.

Reply:
> ![image](https://private-user-images.githubusercontent.com/8814123/370982060-afe91fc0-20d5-4d6e-81e7-f378c12d4562.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjczMjUwMDksIm5iZiI6MTcyNzMyNDcwOSwicGF0aCI6Ii84ODE0MTIzLzM3MDk4MjA2MC1hZmU5MWZjMC0yMGQ1LTRkNmUtODFlNy1mMzc4YzEyZDQ1NjIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkyNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MjZUMDQyNTA5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NWZkNTQ5MzhkN2I3OWQ2MzY0Y2EwMThkZWY3ODMwYjg3YWFmMTYxMDk4NjYyMDA1M2IzMWU4NmE3NzQ3NmYzZCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.O1uckZicieRlw6RW_mEZO9vSCg_jKGdeK0waW6L9tdw) ![image](https://private-user-images.githubusercontent.com/8814123/370982101-7b77a6bd-ddbe-4c6a-bc04-d524572a6619.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjczMjUwMDksIm5iZiI6MTcyNzMyNDcwOSwicGF0aCI6Ii84ODE0MTIzLzM3MDk4MjEwMS03Yjc3YTZiZC1kZGJlLTRjNmEtYmMwNC1kNTI0NTcyYTY2MTkucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkyNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MjZUMDQyNTA5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MzVhYjBhZjk0ZTFkOWE3NTFmODBjNTgzZTY5MTZjOWNkNzQyMjRlMTU0ZTA0MDRiZDU2NDhlY2NmODY0NTQ3YiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.Sa2eRHU5XJ3-e09lHF0AiKgciH_AqQWUvfKoBnqN2F4)
> 
> Section 3.1 in the RFC.

but this is a example, which means that rel is a variable

Reply:
> ![image](https://private-user-images.githubusercontent.com/8814123/370982060-afe91fc0-20d5-4d6e-81e7-f378c12d4562.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjczMjU1MDIsIm5iZiI6MTcyNzMyNTIwMiwicGF0aCI6Ii84ODE0MTIzLzM3MDk4MjA2MC1hZmU5MWZjMC0yMGQ1LTRkNmUtODFlNy1mMzc4YzEyZDQ1NjIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkyNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MjZUMDQzMzIyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZTM2NzZhMmZkOGZhYzk5NzY3M2RkMzI1YjQzMmY5MmY3ZDUwMzNkMzA1NGVkZDRlMjk1NGM5ODk5NWM1YzJiZiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.CWHDwVctO4ZQWlBtG8ZC1Hi9HjA_cK-Mii_HkdoNgZY) ![image](https://private-user-images.githubusercontent.com/8814123/370982101-7b77a6bd-ddbe-4c6a-bc04-d524572a6619.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjczMjU1MDIsIm5iZiI6MTcyNzMyNTIwMiwicGF0aCI6Ii84ODE0MTIzLzM3MDk4MjEwMS03Yjc3YTZiZC1kZGJlLTRjNmEtYmMwNC1kNTI0NTcyYTY2MTkucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkyNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MjZUMDQzMzIyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MTNlYTQ0ZTMxYWIxOTI4NTkwOGMxNDkyMzI5ZGYxNzVmMjRlOWY1OWNlZDg0NDk3ZTNmZWUxNTE0NjdmODRkOSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.p6m5T34VDMTSrVlryxnrJXuX9yUeZ4mjUl02Y3Zfan0)
> 
> Section 3.1 in the RFC.

okay, i understand

Reply:
but i suggest you provide a demo video that you can successfully use webfinger